### PR TITLE
Sub-Site Area review

### PR DESCRIPTION
**Pull request by LucasBrazi06**

### CODE REVIEW

File: src/server/rest/v1/service/UtilsService.ts [hunk 1]
```diff
@@ -2025,4 +2005,99 @@ export default class UtilsService {
     }
     return tag;
   }
+
+  private static buildSiteAreaTree(currentSiteArea: SiteArea, siteAreas: SiteArea[]): SiteArea {
```
Comment (OliveGerste) [inline src/server/rest/v1/service/UtilsService.ts:2009]: This method is also used by the smart charging integration. So I think it should not be placed in UtilsService and should not be private.
I saw that you removed the s for multiple trees from the method name. The reason for the s was that there can be multiple trees on one site. E.G. Site: Mougins --> 3 root site areas --> 3 independent Trees which are build with this method

File: src/server/rest/v1/service/UtilsService.ts [hunk 2]
```diff
@@ -1518,34 +1518,14 @@ export default class UtilsService {
     }
   }
 
-  public static async checkIfSiteAreaTreeValid(siteArea: SiteArea, tenant: Tenant, user: UserToken): Promise<void> {
-    const siteAreas = await SiteAreaStorage.getSiteAreas(tenant,
-      { siteIDs: [siteArea.siteID] }, Constants.DB_PARAMS_MAX_LIMIT,
+  public static async checkIfSiteAreaTreeValid(siteArea: SiteArea): Promise<void> {
+    const siteAreas = await SiteAreaStorage.getSiteAreas(req.tenant,
```
Comment (OliveGerste) [inline src/server/rest/v1/service/UtilsService.ts:1522]: req does not exist in the parameters of the method. This causes build errors.

File: src/server/rest/v1/service/UtilsService.ts [hunk 3]
```diff
@@ -1518,34 +1518,14 @@ export default class UtilsService {
     }
   }
 
-  public static async checkIfSiteAreaTreeValid(siteArea: SiteArea, tenant: Tenant, user: UserToken): Promise<void> {
-    const siteAreas = await SiteAreaStorage.getSiteAreas(tenant,
-      { siteIDs: [siteArea.siteID] }, Constants.DB_PARAMS_MAX_LIMIT,
+  public static async checkIfSiteAreaTreeValid(siteArea: SiteArea): Promise<void> {
```
Comment (OliveGerste) [inline src/server/rest/v1/service/UtilsService.ts:1521]: The changed type definition causes errors in site area service.

**Review by OliveGerste — CHANGES_REQUESTED** [review]
@LucasBrazi06  I made some comments on the PR as I saw that it is to be reviewed.
The check tree method was moved to UtilsService. Do you think the check of the tree will never be needed anywhere else in the application? Maybe at some kind of migration task or something like this, where properties of site areas are changed?

**Comment by LucasBrazi06** [comment]
> @LucasBrazi06 I made some comments on the PR as I saw that it is to be reviewed. The check tree method was moved to UtilsService. Do you think the check of the tree will never be needed anywhere else in the application? Maybe at some kind of migration task or something like this, where properties of site areas are changed?

Not yet to be reviewed, that's my pipeline, otherwise I would have put it in Cross review ;-)
I haven't finished and I'm doing this exercise with Claude who will get back to you to explain/cross check.

**Comment by OliveGerste** [comment]
> > @LucasBrazi06 I made some comments on the PR as I saw that it is to be reviewed. The check tree method was moved to UtilsService. Do you think the check of the tree will never be needed anywhere else in the application? Maybe at some kind of migration task or something like this, where properties of site areas are changed?
> 
> Not yet to be reviewed, that's my pipeline, otherwise I would have put it in Cross review ;-) I haven't finished and I'm doing this exercise with Claude who will get back to you to explain/cross check.

Ah okay. Now I understand. Sorry for my misunderstanding. Let me know if you need anything in this matter.

**Comment by LucasBrazi06** [comment]
Dups, new PR created from another branch

**Pull request closed without merging**